### ISSUE TITLE
Add support for OpenTracing error/message

### ISSUE THREAD

**Issue by vanHoesel** [Perceptyx/perl-redis-opentracing#13]
This module should relay any failures it gets from Redis into the active Span. This helps to detect errors, rather then letting failing spans continue silently.

**Comment by vanHoesel** [comment]
A possible fix would be to wrap [line 79](https://github.com/Perceptyx/perl-redis-opentracing/blob/main/lib/Redis/OpenTracing.pm#L79)
```perl
        return $self->redis->$method_call(@_);
```

 into something like we do in `DBIx::Opentracing`:
[line 359 .. 382](https://github.com/Perceptyx/perl-dbix-opentracing/blob/master/lib/DBIx/OpenTracing.pm#L359-L382)
```perl

        ...
        
        my $result;
        my $wantarray = wantarray;    # eval has its own
        my $failed    = !eval {
            if ($wantarray) {
                $result = [ $handle->$method(@_) ];
            }
            else {
                $result = $handle->$method(@_);
            }
            1;
        };
        my $error = $@;

        if ($failed or defined $handle->err) {
            $span->add_tag(error => 1);
        }
        elsif ($can_count_rows->($handle)) {
            my $rows = sum0(map { $row_counter->($_) } $wantarray ? @$result : $result);
            _add_tag($span, DB_TAG_ROWS,=> $rows);
        }
        $scope->close();

        die $error if $failed;
        return $wantarray ? @$result : $result;
 ```

**Comment by vanHoesel** [comment]
Solved as v0.3.0 of a5a6b68dba7482a77e13389186b1a2faeb2479b0
